insights/specs/datasources/lpstat.py

```diff
@@ -37,7 +37,12 @@ def lpstat_protocol_printers_info(broker):
         for line in content:
             if "device for " in line:
                 "Remove printer address information"
-                result.append(line.rsplit(":", 1)[0])
+                [device, address] = line.split(":", 1)
```

Comment:
How about
```
result.append(line.rsplit(":", 1)[0] if '://' in line else line)
```
?

Reply:
@xiangce - For line "device for test_printer3: socket://192.168.1.5:9100", the result of "result.append(line.rsplit(":", 1)[0] if '://' in line else line)" is "device for test_printer3: socket://192.168.1.5", the address will be returned.

Update to use this: `":".join(line.split(":", 2)[0: 2]) if '://' in line else line`

Reply:
@xiangce - For line "device for test_printer3: socket://192.168.1.5:9100", the result of "result.append(line.rsplit(":", 1)[0] if '://' in line else line)" is "device for test_printer3: socket://192.168.1.5", the address will be returned.

Update to use this: `":".join(line.split(":", 2)[0: 2]) if '://' in line else line`

Reply:
@xiangce - For line "device for test_printer3: socket://192.168.1.5:9100", the result of "result.append(line.rsplit(":", 1)[0] if '://' in line else line)" is "device for test_printer3: socket://192.168.1.5", the address will be returned.

Update to use this: `":".join(line.split(":", 2)[0: 2]) if '://' in line else line`